src/main/java/org/prebid/server/bidder/bidstack/BidstackBidder.java

```diff
@@ -0,0 +1,159 @@
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+public class BidstackBidder implements Bidder<BidRequest> {
+
+    private static final String BIDDER_CURRENCY = "USD";
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    private static final TypeReference<ExtPrebid<?, ExtImpBidstack>> BIDDER_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    public BidstackBidder(String endpointUrl,
+                          CurrencyConversionService currencyConversionService,
+                          JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.currencyConversionService = Objects.requireNonNull(currencyConversionService);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> updatedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final BigDecimal resolvedBidFloor = resolveBidFloor(request, imp);
+                updatedImps.add(updateImp(imp, resolvedBidFloor));
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        final BidRequest updatedBidRequest = updateBidRequest(request, updatedImps);
+
+        return Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(constructHeaders(updatedBidRequest))
+                .body(mapper.encodeToBytes(updatedBidRequest))
+                .payload(updatedBidRequest)
+                .build());
+    }
+
+    private ExtImpBidstack parseExtImp(Imp imp) {
+        try {
+            return mapper.mapper()
+                    .convertValue(imp.getExt(), BIDDER_EXT_TYPE_REFERENCE)
+                    .getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Unable to decode the impression ext for id: " + imp.getId());
+        }
+    }
+
+    private BigDecimal resolveBidFloor(BidRequest request, Imp imp) {
+        return shouldConvertBidFloor(imp.getBidfloor(), imp.getBidfloorcur())
+                ? convertBidFloorCurrency(imp.getBidfloor(), request, imp.getId(), imp.getBidfloorcur())
+                : imp.getBidfloor();
+    }
+
+    private Imp updateImp(Imp imp, BigDecimal bidFloor) {
+        return imp.toBuilder()
+                .bidfloorcur(BIDDER_CURRENCY)
+                .bidfloor(bidFloor)
+                .ext(mapper.mapper().createObjectNode()
+                        .set("bidder", mapper.mapper().valueToTree(parseExtImp(imp))))
```

Comment:
I don't see any similar logic in GO.

Reply:
I added this to get in request such part of the code:
```
        "ext": {
          "bidder": {
            "publisherId": "225a4f40-1df1-48b2-b238-8774a638ce6e"
          }
        }
```

Reply:
It already has that structure.

When you parse `imp.ext` in `parseExtImp`, it deserialize
```
        "ext": {
          "bidder": {
            "publisherId": "225a4f40-1df1-48b2-b238-8774a638ce6e"
          }
        }
```
to `ExtPrebid<?, ExtImpBidstack>` instance.